### PR TITLE
docs: Install -- SAML

### PR DESCRIPTION
**Pull request by seialkali**
Closes  https://jira.d2iq.com/browse/D2IQ-78937 

### Preview

You can preview the docs build using the following URL, adding the PR # where specified:

http://docs-d2iq-com-pr-4164.s3-website-us-west-2.amazonaws.com/

## Checklist

- [ ] Test all commands and procedures, if applicable.
- [ ] Update all links if you are moving a page.
- [ ] Add release date to Release Notes page in the following format: <Package> was released on <Day>, <Month> <Year> Example: `Mesosphere® DC/OS™ 2.1.0 was released on 9, June 2020`

See the [contribution guidelines](https://github.com/mesosphere/dcos-docs-site/blob/main/CONTRIBUTING.md) for more information.

### CODE REVIEW

File: pages/dkp/konvoy/2.1/access-authentication/howto-dex-saml-connector/index.md
Comment (gracedo) [inline pages/dkp/konvoy/2.1/access-authentication/howto-dex-saml-connector/index.md:6]:
```suggestion
excerpt: Connect your Kommander cluster to an Idp using SAML
```

File: pages/dkp/konvoy/2.1/access-authentication/howto-dex-saml-connector/index.md
Comment (gracedo) [inline pages/dkp/konvoy/2.1/access-authentication/howto-dex-saml-connector/index.md:11]:
```suggestion
## How to connect Kommander to an IdP using SAML
```

File: pages/dkp/konvoy/2.1/access-authentication/howto-dex-saml-connector/index.md
Comment (gracedo) [inline pages/dkp/konvoy/2.1/access-authentication/howto-dex-saml-connector/index.md:13]:
```suggestion
This guide shows how to configure your Kommander cluster to use SAML to connect to an identity provider (IdP).
```

File: pages/dkp/konvoy/2.1/access-authentication/howto-dex-saml-connector/index.md
Comment (gracedo) [inline pages/dkp/konvoy/2.1/access-authentication/howto-dex-saml-connector/index.md:15]:
```suggestion
Step 1: [Install Kommander](../../install).
```

**Review by vacas5 — COMMENTED** [review]
I think this might actually belong in the `kommander` folder rather than `konvoy`, if dimitri pointed you to that team for help

**Review by gracedo — COMMENTED** [review]
let's move this into /kommander instead of konvoy. I'll add config updates in my next round of reviews!

File: pages/dkp/konvoy/2.1/access-authentication/howto-dex-saml-connector/index.md
Comment (gracedo) [inline pages/dkp/konvoy/2.1/access-authentication/howto-dex-saml-connector/index.md]:
```suggestion
You can add the configuration as the values field in the `dex` application. An example `dex` configuration provided to the Kommander CLI’s install command would look similar to this:
```

Looked at https://docs.d2iq.com/dkp/kommander/2.1/security/oidc/#add-login-connectors for inspiration

File: pages/dkp/konvoy/2.1/access-authentication/howto-dex-saml-connector/index.md
Comment (gracedo) [inline pages/dkp/konvoy/2.1/access-authentication/howto-dex-saml-connector/index.md]:
```suggestion
apiVersion: config.kommander.mesosphere.io/v1alpha1
kind: Installation
apps:
  dex:
    values: |
      config:
        connectors:
        - type: saml
          id: saml
          name: SAML
          config:
            ssoURL: <url for POST request>
            caData: < base64 PEM encoded CA for the IdP server >
            redirectURI: https://<your-cluster-host>/dex/callback
            entityIssuer: https://<your-cluster-host>/dex
            usernameAttr: < user attribute in saml response >
            emailAttr: < email attribute in saml response >
[...]
```

**Comment by gracedo** [comment]
Actually, this page should go under `/dkp/kommander/2.1/security/`

File: pages/dkp/konvoy/2.1/access-authentication/howto-dex-saml-connector/index.md
Comment (gracedo) [inline pages/dkp/konvoy/2.1/access-authentication/howto-dex-saml-connector/index.md]:
```suggestion
Step 4: Modify the `traefik-foward-auth` configuration and add a whitelist:
```

**Comment by seialkali** [comment]
Made those changes @gracedo! Thanks for the comments.

File: pages/dkp/kommander/2.1/access-authentication/howto-dex-saml-connector/index.md
Comment (gracedo) [inline pages/dkp/kommander/2.1/access-authentication/howto-dex-saml-connector/index.md]:
```suggestion
This step is required to give access to a user to ops-portal. For each user, you must [give access to Kubernetes resources](../rbac) and add an entry in the `whitelist` below.
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (gracedo) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:105]:
```suggestion
Step 5: Run `kommander install` to deploy modified `dex`.
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (gracedo) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:107]:
```suggestion
Step 6: Visit `https://<your-cluster-host>/dkp/kommander/dashboard` to login to the Kommander dashboard.
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (gracedo) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:102]:
```suggestion
apiVersion: config.kommander.mesosphere.io/v1alpha1
kind: Installation
apps:
  traefik-forward-auth-mgmt:
    values: |
      config:
        traefikForwardAuth:
          allowedUser:
            valueFrom:
              secretKeyRef: null
          whitelist:
          - < allowed email addresses >
[...]
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (gracedo) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:83]:
```suggestion
            ssoURL: < url for POST request >
```
tiny nit to make it consistent with the others

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (gracedo) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:94]:
```suggestion
This step is required to give access to a user to the Kommander dashboard. For each user, you must [give access to Kubernetes resources](../rbac) and add an entry in the `whitelist` below.
```

Should we be calling the UI "Kommander dashboard" or something like "web interface" ? (I've seen both)

Comment (seialkali) [reply pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:94]: I can ask in the dkp-ux channel

**Review by gracedo — COMMENTED** [review]
We'll need to also copy this page over to /kommander/2.2

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (Colin1841) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:11]:
```suggestion
## Connect Kommander to an IdP using SAML
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (Colin1841) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:13]:
```suggestion
This section shows how to configure your Kommander cluster to use SAML, so that you can connect to an identity provider (IdP).
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (Colin1841) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:15]:
```suggestion
1.   [Install Kommander](../../install).
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (Colin1841) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:17]:
```suggestion
1.   Configure the IdP
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (Colin1841) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:19]:
```suggestion
Provide the issuer URL and the Assertion Consumer Service (ACS) or callback URL to your IdP.
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (Colin1841) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:21]:
```suggestion
Provide the issuer URL and the Assertion Consumer Service (ACS) or callback URL to your IdP. The issuer URL points to the authentication endpoint at the service provider (Dex), which issues a request towards the IdP via the user agent.
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (Colin1841) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:29]:
```suggestion
The ACS URL points to the service provider (Dex) endpoint that receives SAML assertions issued by the IdP.
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (Colin1841) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:31]:
```suggestion
The ACS or callback URL should look like this:
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (Colin1841) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:37]:
```suggestion
Depending on the IdP, you might be asked to provide the configuration in some form of an XML snippet - see the following example:
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (Colin1841) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:49]:
```suggestion
1.   Modify the `dex` configuration:
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (Colin1841) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:51]:
```suggestion
For this step, obtain the following from your IdP:
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (Colin1841) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:63]:
```suggestion
Ensure you base64 encode the contents of the PEM file. As an example, the prefix of the contents will result into this exact base64 prefix:
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (Colin1841) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:69]:
```suggestion
You can add the configuration as the values field in the `dex` application. An example `dex` configuration provided to the Kommander CLI’s install command should look similar to:
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (Colin1841) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:92]:
```suggestion
1.   Modify the `traefik-foward-auth-mgmt` configuration and add a whitelist:
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (Colin1841) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:111]:
```suggestion
1.   Run `kommander install` to deploy modified `dex`.
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (Colin1841) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:115]:
```suggestion
1.   Visit `https://<your-cluster-host>/dkp/kommander/dashboard` to login to the Kommander dashboard.

1.   Select `Launch Console` and follow the authentication steps to complete the procedure.
```

**Review by Colin1841 — APPROVED** [review]

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (seialkali) [reply pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:17]: Just to double check, did you mean `2.` here instead of `1.`?

Comment (seialkali) [reply pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:17]: Actually, I've just noticed that all the steps have been changed to `1.`. Can I ask why that is?

Comment (gracedo) [reply pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:17]: In markdown you can number all steps with `1.`'s which automatically renders as an ordered list, which is nice so if you ever had to add a step in the middle you don't need to redo the entire numbered list https://www.markdownguide.org/basic-syntax/#ordered-lists

Comment (seialkali) [reply pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:17]: Good to know! However, locally, it's not working as expected. The first two steps are numbered correctly, but the 3rd step is resetting back to 1. I'm guessing because of all the copy in between the steps. Not sure if there's a way around this or if we need to go with a different approach.

Comment (gracedo) [reply pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:17]: Probably something to do with the indents and spacing to get it working properly. I can take a look at your branch to try to get it working

Comment (seialkali) [reply pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:17]: That's alright, I'm sure you've got loads of your own work! You've already been super helpful. I can play around with it on Monday and see if I sort it out. Thanks for the offer though!

Comment (gvasquezvargas) [reply pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:17]: @seialkali, could you please let me know if you find a solution? I am currently working on a document where I am seeing the same behavior.

**Comment by msdolbey** [comment]
@seialkali - I made a big commit with lots of linting changes - but not really any content. You'll want to pull before you push any changes.

**Comment by seialkali** [comment]
@msdolbey Thanks for the heads up!

**Comment by seialkali** [comment]
@msdolbey This page links out to an RBAC page which hasn't been created yet. Should I hold off on merging until that page has been made? There's a ticket for the RBAC page here: https://jira.d2iq.com/browse/D2IQ-78938

**Review by msdolbey — APPROVED** [review]
👍 

Looks good to me @seialkali 
I added a bit about using the kommander install config file:

```
kommander install --installer-config kommander.yaml
```

I followed the steps, but didn't spin up my own SAML server to test end-to-end, so I'm not 100% that this is right, but it really seemed to work for me. I mean, especially because following these steps I messed up my login on the cluster because I put in some garbage there for the server info.
But, might want someone from @mesosphere/kommander to double check everything here, but looks good to me.

credentials

**Comment by msdolbey** [comment]
@seialkali - for that RBAC page - 
This was put here:
https://docs.d2iq.com/dkp/kommander/2.1/operations/access-control/rbac/
So, I think you're okay with that. 

That [ticket](https://jira.d2iq.com/browse/D2IQ-78938) is probably ready to get revisited and you can assign to me if you want.

Oh, before merging - let's copy this into 2.2, one moment

**Comment by msdolbey** [comment]
@seialkali - alrighty! done.

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (seialkali) [reply pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md]: @gvasquezvargas should I just use `Kommander UI` here?

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (seialkali) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:116]: @msdolbey how come this has been add to the bottom of the page?

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (gvasquezvargas) [reply pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md]: I guess it depends. Is the UI's wording already been changed to DKP? If yes, I would rather change it to DKP UI already, but if it still called Kommander, I would leave it that way. @cmastr what do you think?

Comment (seialkali) [reply pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md]: Changing this to `DKP UI` based on this Slack thread, but if this is wrong for some reason, let me know: https://mesosphere.slack.com/archives/C052HL47U/p1647015731820569

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (cmastr) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:35]:
```suggestion
    Depending on the IdP, you may need to provide the configuration in an XML snippet. See the following example, making sure to replace `<your-cluster-host>` with your URL:
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (cmastr) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:47]:
```suggestion
1.  Edit the `dex` configuration:
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (cmastr) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:67]:
```suggestion
    You can add the configuration as the values field in the `dex` application. An example `dex` configuration provided to the [Kommander CLI’s install command][kommander-install] looks like:
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (cmastr) [inline pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:90]:
```suggestion
1.  Edit the `traefik-foward-auth-mgmt` configuration and add a whitelist:
```

File: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md
Comment (cmastr) [reply pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:116]: Brittany this is the way we do links. So the link above in the sentence links to the bottom of the page, then the bottom of the page links to the actual page. That way we can reuse links throughout a document without duplicating efforts.

Comment (cmastr) [reply pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md:116]: I see now there are some incorrectly formatted links in this document.

File: pages/dkp/kommander/2.2/security/howto-dex-saml-connector/index.md
Comment (cmastr) [inline pages/dkp/kommander/2.2/security/howto-dex-saml-connector/index.md:49]:
```suggestion
    For this step, get the following from your IdP:
```

**Review by cmastr — APPROVED** [review]

**Review by cneth — APPROVED** [review]

**Pull request merged**
